### PR TITLE
Update release process instruction.

### LINKED ISSUE #9718: Docs updates needed
- release dependencies:
   - like keyring
   - testpath
   - graphviz

- say env variable are **examples**

- report to github that 2fa + App api does not return OTP needed in headers. 
- tools/stat should print/ask password on stderrr. 
- say to update `whatsnew/index.rst` with the stats/ remove/ re add developpement
- generating stats should get a header/link or the toctree will be empty
- release-upload should not rebuild
- move twine upload on its own block.
- update readthedocs.

Will update this issue while we discover more issues during release process.

### PR DESCRIPTION
**Pull request by Carreau**
According to what's in #9718, that should make things a bit more
organized.

Closes #9718

### CODE REVIEW

File: docs/source/coredev/release_process.rst [hunk 1]
```diff
@@ -72,6 +80,9 @@ If a major release:
       file you just created and remove temporarily the first entry called
       ``development`` (you'll need to add it back after release).
 
+      Make sure that the stats file have a header or it won't be rendered in
```
Comment (willingc) [inline docs/source/coredev/release_process.rst]: have -> has

File: docs/source/coredev/release_process.rst [hunk 2]
```diff
@@ -216,6 +229,13 @@ The IPython website should document the new release:
 - update current version and download links
 - update links on the documentation page (especially if a major release)
 
+12. Update readthedocs
+----------------------
+
+Make sure to update readthedocs and set the latest tag as stable, as well as
+checked that previous release is still building under its own tag.
```
Comment (willingc) [inline docs/source/coredev/release_process.rst]: checked that -> checking that the

**Comment by willingc** [comment]
@carreau A couple of minor edits and this looks good to me. You may want another person to do a quick review of the script to make sure that I didn't miss any typos.

File: docs/source/coredev/release_process.rst [hunk 3]
```diff
@@ -8,7 +8,15 @@ This document contains the process that is used to create an IPython release.
 
 Conveniently, the ``release`` script in the ``tools`` directory of the ``IPython``
 repository automates most of the release process. This document serves as a
-handy reminder and checklist for the release manager.
+handy reminder and checklist for the release manager. 
+
+During the release process, you might need the extra following dependencies:
+
+ - ``keyring`` to access your GitHub authentication tokens
+ - ``testpath`` to fake some file system operations
```
Comment (takluyver) [inline docs/source/coredev/release_process.rst]: Isn't testpath just a regular test requirement? It seems a bit odd to list it specifically here; I don't think it's directly involved in the release process.

File: docs/source/coredev/release_process.rst [hunk 4]
```diff
@@ -216,6 +229,13 @@ The IPython website should document the new release:
...
 - update links on the documentation page (especially if a major release)
 
+12. Update readthedocs
+----------------------
+
+Make sure to update readthedocs and set the latest tag as stable, as well as
+checked that previous release is still building under its own tag. 
+
+
 12. Celebrate!
```
Comment (takluyver) [inline docs/source/coredev/release_process.rst]: This should probably be 13 now. Unless we think that's unlucky ;-)

File: tools/release [hunk 5]
```diff
@@ -45,44 +45,45 @@ print()
 
 # Perform local backup, go to tools dir to run it.
 cd(tooldir)
-sh('./make_tarball.py')
-sh('mv ipython-*.tgz %s' % ipbackupdir)
 
-# Build release files
-sh('./build_release %s' % ipdir)
+if 'upload' in sys.argv:
```
Comment (takluyver) [inline tools/release]: This looks like we now have essentially two scripts in one file. Would it be tidier to separate them?

Comment (Carreau) [inline tools/release]: Unsure it is worth it. We will likely forget to update common variables if we split it.

File: docs/source/coredev/release_process.rst (hunk 4 above)
Comment (Carreau) [inline docs/source/coredev/release_process.rst]: Done.

File: docs/source/coredev/release_process.rst (hunk 3 above)
Comment (Carreau) [inline docs/source/coredev/release_process.rst]: Yes you are right, just filled that form the note I took was @fperez was releasing 5.0 and testpath was one of the missing packages.

**Pull request merged**